src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -723,4 +643,98 @@ public static class Defaults {
             FIELD_TYPE.freeze();
         }
     }
+
+    // Helper class used to validate builder before build is called. Needs to be invoked in 2 places: during
+    // parsing and during merge.
+    private static class BuilderValidator {
```

Comment:
Can we have this class in Mapper folder as package private , just to clean this file.